README.md

```diff
@@ -20,38 +23,131 @@ var config = {
 };
 
 const client = new ag.Client(config);
+client.connect();
+client.end();
+```
+Use client.end() to terminate the connection to the database. Calling client.end() without a callback function will return a promise.
+
+## Queries
+To retrieve data from AgensGraph, use the query() API function to query the database. There are multiple ways to send queries, including passing text directly, using parameters, passing in objects, and using prepared statements. The following examples illustrate the different possible methods of querying.
+
+Users can retrieve Cypher data by passing text directly to client.query():
+```js
+var ag = require('agensgraph');
+const client = new ag.Client(Config);
+
+client.connect();
+client.query('SET graph_path = gpt'); //set graph path to gpt
+client.query("MATCH (a:person)-[r:knows]->(b:person) WHERE b.name = 'Adam' RETURN a,r,b;", function(err, res){
+    if (err) {
+      throw err;
+    } else {
+      console.log(res.rows);
+    }
+});
+client.end();
+```
+Users can also send parameterized queries:
+```js
+...
+const text = "MATCH (a:person)-[:knows]->(b:person) WHERE b.name = $1 RETURN a;"
+const values = ['Adam'];
+
+client.connect();
+client.query('SET graph_path = gpt'); //set graph path to gpt
+client.query(text, values, function(err, res){
+    if (err) {
+      throw err;
+    } else {
+      console.log(res.rows);
+    }
+});
+```
+Passing a config object:
+```js
+const query= {
+  text: "MATCH (a:person)-[:knows]->(b:person) WHERE b.name = $1 RETURN a;",
+  values = ['Adam'];
+  }
+
+client.connect();
+client.query('SET graph_path = gpt'); //set graph path to gpt
+client.query(query, function(err, res){
+    if (err) {
+      throw err;
+    } else {
+      console.log(res.rows);
+    }
+});
+```
+Users can also use named prepared statements. Using prepared statements allows query execution plans to be cached on the server on a per connection statement. Once a query name is sent to the server, it will be parsed and executed. On subsequent sends the query will skip the parse request and instead send the name of the query:
+```js
+...
+const query = {
+  name: 'fetch-data',
+  text: "MATCH(n) WHERE n.name = 'Eve' RETURN n"
+}
+
+client.query(query, (err,res) => {
+  if (err) {
+    throw err;
+  } else {
+    console.log(res.rows);
+  }
+});
+
+```
+
+## Creating Data
+To create data in AgensGraph, the query() API function is again used to query the database. Below is an example of creating Cypher data by passing text directly to client.query():
+```js
+var ag = require('agensgraph');
+const client = new ag.Client(Config);
 
-client.query('DROP GRAPH IF EXISTS gpt CASCADE');
+client.connect();
 client.query('CREATE GRAPH gpt');
 client.query('SET graph_path = gpt');
+client.query("CREATE (a:person {name:'Alice')-[r:knows {from: 'school'}]->(b:person {name: 'Bob'} RETURN a,r,b)", function (err, res) {
```

Comment:
CREATE statement appears to have a syntax error.

Reply:
Addressed this error